validator/client/validator.go

```diff
@@ -1139,7 +1140,40 @@ func (v *validator) StartEventStream(ctx context.Context, topics []string, event
 	v.validatorClient.StartEventStream(ctx, topics, eventsChannel)
 }
 
-func (v *validator) ProcessEvent(event *eventClient.Event) {
+func (v *validator) checkDependentRoots(ctx context.Context, head *structs.HeadEvent, slot primitives.Slot) error {
```

Comment:
this is nice, gRPC matches rest events API  by overloading the stream slots grpc stream. great work.